File: src/app/core/models/move-task-in-group-request.ts
import { TaskViewModel } from './view-models/project-task-dto';

export interface MoveTaskInGroupRequest {
  taskId: number;
  tasks: TaskViewModel[];
  newGroupId: number;
  oldGroupId: number;
  sortOrder?: number;
  previousIndex: number;
  currentIndex: number;
}

File: Netptune.Core/Requests/MoveTaskInGroupRequest.cs
namespace Netptune.Core.Requests
{
    public class MoveTaskInGroupRequest
    {
        public int TaskId { get; set; }

        public int NewGroupId { get; set; }

        public int OldGroupId { get; set; }

        public double? SortOrder { get; set; }

        public int PreviousIndex { get; set; }

        public int CurrentIndex { get; set; }
    }
}
